disvae/losses.py

```diff
@@ -229,3 +325,32 @@ def _kl_normal_loss(mean, logvar, storer=None):
         for i in range(latent_dim):
             storer['kl_loss_' + str(i)] += latent_kl[i].item()
     return total_kl
+
+
+def _permute_dims(latent_dist):
+    """
+    Implementation of Algorithm 1 in ref [1]. Randomly permutes the sample from
+    q(z) (latent_dist) across the batch for each of the latent dimensions (mean
+    and log_var).
+
+    Parameters
+    ----------
+    latent_dist : torch.Tensor
+        sufficient statistics of the latent dimension. E.g. for gaussian
+        (mean, log_var), each of shape : (batch_size, latent_dim).
+
+    References :
+        [1] Kim, Hyunjik, and Andriy Mnih. "Disentangling by factorising."
+        arXiv preprint arXiv:1802.05983 (2018).
+
+    """
+
+    perm = torch.zeros(latent_dist.size())
```

Comment:
did you test on GPU? suprised it works! It seem to me that perm is still on cpu. YOu should do `torch.zeros_like(latent_dist)`

Reply:
Good point, I'll change this